src/threadtracer.c

```diff
@@ -49,10 +49,13 @@ static const char *threadnames[MAXTHREADS];
 //! The thread-ids.
 static pthread_t threadids[MAXTHREADS];
 
+__thread int tidx;
```

Comment:
The introduction of thread-local storage is pretty well. Thanks for pointing this. However, to eliminate symbol exposure, add `static` for the variable.